dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:meta/meta.dart';
+import 'package:stack_trace/stack_trace.dart';
+
+import 'protocol/noop_origin.dart'
+    if (dart.library.html) 'protocol/origin.dart';
+import 'protocol/sentry_stack_frame.dart';
+import 'sentry_options.dart';
+
+/// converts [StackTrace] to [SentryStackFrames]
+class SentryStackTraceFactory {
+  /// A list of string prefixes of module names that do not belong to the app, but rather third-party
+  /// packages. Modules considered not to be part of the app will be hidden from stack traces by
+  /// default.
+  final List<String> _inAppExcludes;
+
+  /// A list of string prefixes of module names that belong to the app. This option takes precedence
+  /// over inAppExcludes.
+  final List<String> _inAppIncludes;
+
+  SentryStackTraceFactory(SentryOptions options)
+      : _inAppExcludes = options?.inAppExcludes,
+        _inAppIncludes = options?.inAppIncludes;
+
+  /// returns the [SentryStackFrame] list from a stackTrace ([StackTrace] or [String])
+  List<SentryStackFrame> getStackFrames(dynamic stackTrace) {
+    if (stackTrace == null) return null;
+
+    final chain = stackTrace is StackTrace
+        ? Chain.forTrace(stackTrace)
+        : Chain.parse(stackTrace as String);
+
+    final frames = <SentryStackFrame>[];
+    for (var t = 0; t < chain.traces.length; t += 1) {
+      final encodedFrames =
+          chain.traces[t].frames.map((f) => encodeStackTraceFrame(f));
+
+      frames.addAll(encodedFrames);
+
+      if (t < chain.traces.length - 1) {
+        frames.add(SentryStackFrame.asynchronousGapFrameJson);
+      }
+    }
+
+    return frames.reversed.toList();
+  }
+
+  /// converts [Frame] to [SentryStackFrame]
+  @visibleForTesting
+  SentryStackFrame encodeStackTraceFrame(Frame frame) {
+    final fileName =
+        frame.uri.pathSegments.isNotEmpty ? frame.uri.pathSegments.last : null;
+
+    final sentryStackFrame = SentryStackFrame(
+      absPath: '$eventOrigin${_absolutePathForCrashReport(frame)}',
+      function: frame.member,
+      lineNo: frame.line,
+      colNo: frame.column,
+      inApp: isInApp(frame),
+      fileName: fileName,
+      package: frame.package,
+      // TODO platform, postContext, preContext, rawFunction,
+      // TODO ? module, native, frame, contextLine, framesOmitted, imageAddr,... ?
+      // module: frame.library,
```

Comment:
is this a TODO or the language doesn't support those things? I'd change the comment if so

Reply:
it's more a question, in java, the `getStackFrames` doesn't set all this fields, so is it to do ?

Reply:
only if this exists in the language, the protocol is big enough to support many languages, if its not the case in Dart, we could even remove the comments

Reply:
✅  i removed the comment